core/variant/type_info.h

```diff
@@ -220,31 +226,47 @@ inline String enum_qualified_name_to_class_info_name(const String &p_qualified_n
 	// Contains namespace. We only want the class and enum names.
 	return parts[parts.size() - 2] + "." + parts[parts.size() - 1];
 }
+
+template <typename T>
+inline String demangle_enum_name() {
+#ifdef __GNUC__
+	int status;
+	char *buffer = abi::__cxa_demangle(typeid(T).name(), nullptr, nullptr, &status);
+	DEV_ASSERT(status == 0);
+	String name = String::utf8(const_cast<const char *>(buffer));
+	free(buffer);
+	return name;
+#else
+	return String::utf8(typeid(T).name()).trim_prefix("enum ");
```

Comment:
I guess this case is valid for both MSVC and emscripten? Do they not mangle names? Do we have any guarantees that this works for _potential_ other compilers?

Reply:
GCC and Clang are the main two compilers that return mangled names from `typeid`[^1]. I *believe* emscription falls into the same camp as clang, but I'm not 100% positive.

[^1]: https://en.cppreference.com/w/cpp/types/type_info/name#Notes